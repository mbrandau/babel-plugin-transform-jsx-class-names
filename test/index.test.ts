import { transform } from '@babel/core';
import plugin, { Options } from '../src';

const example = `
function classNames() {
  return args.filter(a=>!!a).join(' ')
}
const cls = cn("text-blue")
function Component(props) {
  let randomClass = ["container"]
  randomClass.push("text-red")
  return <div className="container">
    <span className={classNames("text-red",{"container":false},randomClass,\`container \${randomClass}\`,cn('text-blue'))}>{props.children}</span>
  </div>;
}`;

const replaceClassName = (className: string) => `replaced_${className}`;

function runTest(code: string, pluginConfig?: Options) {
  return expect(
    transform(code, {
      plugins: [[plugin, { replaceClassName, ...(pluginConfig ?? {}) }]],
    })!.code
  ).toMatchSnapshot();
}

it('basic', () => {
  runTest(example);
});

it('experimentalReplaceVariables', () =>
  runTest(example, {
    experimentalReplaceVariables: true,
  }));

it('replacerFunctionName', () =>
  runTest(example, {
    replacerFunctionName: 'cn',
  }));
