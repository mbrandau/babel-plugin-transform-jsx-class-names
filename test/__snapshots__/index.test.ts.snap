// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic 1`] = `
"function classNames() {
  return args.filter(a => !!a).join(' ');
}

const cls = cn(\\"text-blue\\");

function Component(props) {
  let randomClass = [\\"container\\"];
  randomClass.push(\\"text-red\\");
  return <div className=\\"replaced_container\\">
    <span className={classNames(\\"replaced_text-red\\", {
      \\"replaced_container\\": false
    }, randomClass, \`replaced_container \${randomClass}\`, cn(\\"replaced_text-blue\\"))}>{props.children}</span>
  </div>;
}"
`;

exports[`experimentalReplaceVariables 1`] = `
"function classNames() {
  return args.filter(a => !!a).join(' ');
}

const cls = cn(\\"text-blue\\");

function Component(props) {
  let randomClass = [\\"replaced_container\\"];
  randomClass.push(\\"text-red\\");
  return <div className=\\"replaced_container\\">
    <span className={classNames(\\"replaced_text-red\\", {
      \\"replaced_container\\": false
    }, randomClass, \`replaced_container \${randomClass}\`, cn(\\"replaced_text-blue\\"))}>{props.children}</span>
  </div>;
}"
`;

exports[`replacerFunctionName 1`] = `
"function classNames() {
  return args.filter(a => !!a).join(' ');
}

const cls = cn(\\"replaced_text-blue\\");

function Component(props) {
  let randomClass = [\\"container\\"];
  randomClass.push(\\"text-red\\");
  return <div className=\\"replaced_container\\">
    <span className={classNames(\\"replaced_text-red\\", {
      \\"replaced_container\\": false
    }, randomClass, \`replaced_container \${randomClass}\`, cn(\\"replaced_text-blue\\"))}>{props.children}</span>
  </div>;
}"
`;
